Dockerfile

```diff
@@ -18,15 +18,35 @@ FROM openjdk:8-jre-alpine
 
 LABEL maintainer="kezhenxu94@apache.org"
 
-EXPOSE 8080 11800 12800
+ENV SW_HOME=/skywalking
+ENV ZK_HOME=/zk
+ENV ES_HOME=/es
+
+# MODE: standalone/cluster
+# standalone: will start only one OAP server
+# cluster: will start a cluster coordinator (ZK) and multiple OAP servers
+ENV MODE=standalone
+
+EXPOSE 8081 8082
+EXPOSE 11800 12800 11801 12801
+EXPOSE 2181 9200
 EXPOSE 9090 9091 9092 9093 9094
 
-ENV SW_HOME=/skywalking
+RUN apk update && apk add bash
 
 VOLUME $SW_HOME
+VOLUME $ZK_HOME
+VOLUME $ES_HOME
```

Comment:
I would suggest you move ZK and ES download as part of CI, rather than built inside the image. The reason is, thinking in this way, if you try to test `etcd` cluster management, then what happens? Then tests more on dynamic configuration, alarm, more cluster management implementation or more storage implementation, what happen?

The e2e container should be a container only, we could allow which case do preparation after container started, rather than build a large size container image.

Reply:
> I would suggest you move ZK and ES download as part of CI, rather than built inside the image.

I had thought about this once, the reason why I give up doing this is the fact that we have to download them every time we run a test, what's worse, the download links may be unavailable in the future (maybe because its version will become old that it isn't supported officially?)

> The reason is, thinking in this way, if you try to test etcd cluster management, then what happens? Then tests more on dynamic configuration, alarm, more cluster management implementation or more storage implementation, what happen?

> The e2e container should be a container only, we could allow which case do preparation after container started, rather than build a large size container image.

This should be reasonable to keep it just a container and provide possibilities for the test to do the preparation works.

Anyway, since the bandwidth is not a bottleneck (for now) and the download links of zk and es are still available, I think it's ok to follow what you said for the sake of extensibility, thanks a ton.